homeassistant/components/mobile_app/__init__.py

```diff
@@ -124,12 +124,18 @@ async def manage_cloudhook(state: cloud.CloudConnectionState) -> None:
         ):
             await async_create_cloud_hook(hass, webhook_id, entry)
 
-    if (
-        CONF_CLOUDHOOK_URL not in entry.data
-        and cloud.async_active_subscription(hass)
-        and cloud.async_is_connected(hass)
-    ):
-        await async_create_cloud_hook(hass, webhook_id, entry)
+    if cloud.async_is_logged_in(hass):
+        if (
+            CONF_CLOUDHOOK_URL not in entry.data
+            and cloud.async_active_subscription(hass)
+            and cloud.async_is_connected(hass)
```

Comment:
So, if there is a temporary hickup (e.g., connection issue, payment issue and what not), it will end up loosing all cloud hooks?